app/sliders/plotter.py

```diff
@@ -237,6 +237,87 @@ def scale_values(self, values, max_value=150):
         return scaled_values
 
 
+class InteractiveBubblePlotter(InteractivePlotter):
+    def __init__(self, raw_data, config):
+        super().__init__(raw_data, config)
+
+        self.filters_single_choice = None
+
+    def fit_data(self):
+        # TODO: I can probably initialize a dict already in the abstract class for self.fitted_data.
+        #  Also when creating plots. Do this in the refactoring phase.
+        self.fitted_data = {}
+
+        for code in self.config["plot_codes"]:
+            single_choice_dict = self.raw_data[code][self.config["filters"]["single_choice_default"]]
+            source = ColumnDataSource(data={"x": single_choice_dict["x"],
+                                            "y": single_choice_dict["y"],
+                                            "z": self.scale_values(single_choice_dict["z"]),
+                                            "color": [n for n in range(len(single_choice_dict["x"]))],
+                                            "labels": single_choice_dict["labels"]})
+            self.fitted_data[code] = source
+
+    def create_plot(self):
+        self.plot = {}
+
+        for code in self.config["plot_codes"]:
+            # Get x and y range
+            x_range = []
+            y_range = []
+
+            for key in self.config["filters"]["single_choice"]:
+                x_range.extend(self.raw_data[code][key]["x"])
+                y_range.extend(self.raw_data[code][key]["y"])
+
+            # Create the figure
+            plot = figure(**self.config["figure"],
+                          x_range=[min(x_range), max(x_range)],
+                          y_range=[min(y_range), max(y_range)])
+
+            # Add circles to the plot
+            mapper = linear_cmap(field_name="color", palette=Category20[20], low=0, high=20)
+            plot.circle(x="x", y="y", size="z", color=mapper, source=self.fitted_data[code], legend_group="labels")
+
+            # Set the position of the legend
+            plot.add_layout(plot.legend[0], "right")
+
+            self.plot[code] = plot
+
+    def scale_values(self, values, max_value=150):
+        """
+        Helper function.
+        Scale values of a list so that they don't exceed a maximum.
+
+        :param values: list, containing integers
+        :param max_value: int, maximum that the values shouldn't exceed
+        :return: list, scaled values
+        """
+        max_val = max(values)
+        scaling_factor = max_value / max_val if max_val > max_value else 1.0
```

Comment:
Small thing: I wonder if there’s a clearer naming instead of having `max_value` and `max_val` here (which is a bit confusing).

Reply:
That's true, will find a better naming 👀